File: JavaScript/Gate Four/Level8Task.js

function getPrimes(numbers) {
    let temp = new Array(numbers.length);
    let count = 0;

    for (let counter = 0; counter < numbers.length; counter++) {
        let number = numbers[counter];

        if (number > 1) {
            let isPrime = true;

            for (let divisor = 2; divisor < number; divisor++) {
            if (number % divisor === 0) {
                    isPrime = false;
    }
      }

            if (isPrime) {
                temp[count] = number;
                count++;
            }
        }
    }

    let result = [];
    for (let counter = 0; counter < count; counter++) {
        result[counter] = temp[counter];
    }

    return result;
}






function replaceNegatives(numbers) {
    for (let counter = 0; counter < numbers.length; counter++) {
        if (numbers[counter] < 0) {
            numbers[counter] = 0;
        }
    }
    return numbers;
}


function findDuplicates(numbers) {
    let temp = new Array(numbers.length);
    let count = 0;

    for (let index = 0; index < numbers.length; index++) {
        for (let nextIndex = index + 1; nextIndex < numbers.length; nextIndex++) {

            if (numbers[index] === numbers[nextIndex]) {

             let alreadyExists = false;

              for (let checkIndex = 0; checkIndex < count; checkIndex++) {
                 if (temp[checkIndex] === numbers[index]) {
                    alreadyExists = true;
       }
              }

                if (!alreadyExists) {
                    temp[count] = numbers[index];
                    count++;
 }
  }
        }
    }

    let result = [];
    for (let resultIndex = 0; resultIndex < count; resultIndex++) {
        result[resultIndex] = temp[resultIndex];
    }

    return result;
}



function moveZeros(numbers) {
    let result = new Array(numbers.length);
    let position = 0;

    for (let counter = 0; counter < numbers.length; counter++) {
        if (numbers[i] !== 0) {
            result[position] = numbers[i];
            position++;
        }
    }

  
    for (let counter = position; counter < numbers.length; counter++) {
        result[i] = 0;
    }

    return result;
}

File: Java/Gate Four/Level8Task.java
import java.util.Arrays;

public class Level8Task {

    public static int [] getPrimes(int [] numbers) {
        int [] temp = new int[numbers.length];
        int count = 0;

        for (int number : numbers) {
            if (number > 1) {
                boolean isPrime = true;

                for (int divisor = 2; divisor * divisor <= number; divisor++) {
                    if (number % divisor == 0) {
                        isPrime = false;
                        break;
       }
                }

                if (isPrime) {
                    temp[count++] = number;
      }
            }
        }

        int [] result = new int[count];
        for (int index = 0; index < count; index++) {
            result[index] = temp[index];
        }

        return result;
    }

    public static int [] replaceNegatives(int[] numbers) {
        for (int index = 0; index < numbers.length; index++) {
            if (numbers[index] < 0) {
                numbers[index] = 0;
            }
        }
        return numbers;
    }

    public static int [] moveZeros(int[] numbers) {
        int[] result = new int[numbers.length];
        int position = 0;

        for (int number : numbers) {
            if (number != 0) {
                result[position++] = number;
            }
        }

        return result;
    }

    public static int[] findDuplicates(int[] numbers) {
        int[] temp = new int[numbers.length];
        int count = 0;

        for (int currentIndex = 0; currentIndex < numbers.length; currentIndex++) {
            for (int nextIndex = currentIndex + 1; nextIndex < numbers.length; nextIndex++) {

                if (numbers[currentIndex] == numbers[nextIndex]) {

                    boolean alreadyExists = false;

                    for (int checkIndex = 0; checkIndex < count; checkIndex++) {
                        if (temp[checkIndex] == numbers[currentIndex]) {
                            alreadyExists = true;
                            break;
                 }
                    }

                    if (!alreadyExists) {
                        temp[count++] = numbers[currentIndex];
             }
  }
      }
        }

     
        int[] result = new int[count];
        for (int index = 0; index < count; index++) {
            result[index] = temp[index];
        }

        return result;
    }

    public static void main(String[] args) {
        int[] numbers = {5, 9, 3, 6, 2, 0, -1, 3};

        System.out.println(Arrays.toString(getPrimes(numbers)));
        System.out.println(Arrays.toString(replaceNegatives(numbers)));
        System.out.println(Arrays.toString(moveZeros(numbers)));
        System.out.println(Arrays.toString(findDuplicates(numbers)));
    }
}
